### PR TITLE
fixed illegalthreadstateexception: ontick event is run via platform.r…

### PR DESCRIPTION
**Pull request by sebasv**
…unlater and ticktest is split up in multiple tests that also test more cases.

### CODE REVIEW

**Comment by sebasv** [comment]
I used sleep(1550) to make extra sure that the timing is correct, even on the slower Travis environment.  I can't be sure whether it is necessary, but that's irrelevant for the testcases. Only the diff test coverage failed since I didn't write any new lines that are testable ;)

**Comment by karelvandoesburg** [comment]
looks good. just for my own knowledge, what does making the method synchronized improve or do?

**Comment by sebasv** [comment]
Basically, It ensures that the code in the method is actually properly executed. If the tick method wouldn't be synchronized, then the code in the method would be executed in the thread of the caller. However, this thread might have his own buffer for the tick fields. So basically in a non-synchronized method call you don't know whether you change the state of the tick object that resides in the tick thread, or you change the state of the `shadow' tick object in the current thread.

File: src/main/java/tetris/Controller.java [hunk 1]
```diff
@@ -27,8 +27,8 @@
     private boolean gameOver = false;
     private TetrominoMovementHandler movementHandler = new TetrominoMovementHandler(this);
     private Tick timer = new Tick(event -> {
-        movementHandler.lowerTetromino(tetromino, grid);
-        redraw();
+        Platform.runLater(() -> movementHandler.lowerTetromino(tetromino, grid));
```
Comment (karelvandoesburg) [inline src/main/java/tetris/Controller.java]: Nice, good to see that what we discussed in today's meeting works

**Comment by karelvandoesburg** [comment]
is it possible to get the last check covered with a test as well? Or is this the part that's practically not testable?

**Comment by sebasv** [comment]
Can you be more specific? Which last check do you mean?

**Comment by karelvandoesburg** [comment]
line 95 and 96 in the tick still need to be covered, the uncovered controller code will be tested by me later this week

**Comment by sebasv** [comment]
Your wish...

**Comment by karelvandoesburg** [comment]
Nice, i like it

**Pull request merged**